shardmanager.go

```diff
@@ -47,7 +46,7 @@ func getCollationsTopic(shardID ShardIDType) string {
 func NewShardManager(ctx context.Context, h host.Host, eventNotifier EventNotifier) *ShardManager {
 	service, err := pubsub.NewGossipSub(ctx, h)
 	if err != nil {
-		log.Fatalln(err)
```

Comment:
One question for using `Fatal` or `Panic`, how do you choose which to use in the code? It seems to me that you use them in the same conditions.